discovery-provider/src/api/v1/tracks.py

```diff
@@ -130,6 +155,49 @@ def get(self):
         return success_response(tracks)
 
 
+# Trending
+#
+# There are two trending endpoints - regular and full. Regular
+# uses the familiar caching decorator, while full is more interesting.
+#
+# Full Trending is consumed page by page in the client, but we'd like to avoid caching
+# each page seperately (to avoid old pages interleaving with new ones on the client).
+# We're further constrained by the need to fetch more than the page size of ~10 in our playcount
+# query in order to score + sort the tracks.
+#
+# We address this by always fetching and scoring `TRENDING_LIMIT` (>> page limit) tracks,
+# caching the entire tracks list. This cached value is sliced by limit + offset and returned.
+# This cache entry is be keyed by genre + user_id + time_range.
+#
+# However, this causes an issue where every distinct user_id (every logged in user) will have a cache miss
+# on their first call to trending. We deal with this by adding an additional layer of caching inside
+# `get_trending_tracks.py`, which caches the scored tracks before they are populated (keyed by genre + time).
+# With this second cache, each user_id can reuse on the same cached list of tracks, and then populate them uniquely.
+
+TRENDING_LIMIT = 100
+TRENDING_TTL_SEC = 30 * 60
+
+def get_trending(args):
+    """Get Trending, shared between full and regular endpoints."""
+    # construct args
+    time = args.get("time") if args.get("time") is not None else 'week'
```

Comment:
could be args.get("time", "week") right?

Reply:
Oh yes definitely, I just blindly copypasted this from the old one. Will change

Reply:
leaving as is due to strange behavior of get, which will return None if the value for key is None